### ISSUE TITLE
Remove hard-coded feed IDs from Calf ration Routines

### ISSUE THREAD

**Issue by KFosterReed** [RuminantFarmSystems/RuFaS#2078]
## Expected Behavior

The feeds IDs used to determine the available feeds for the calf diet formulation should be determined by the user, not hard coded 

## Current Behavior

There are currently 2 places where the calf ration formulation relies on hard-coded values:

1. In `get_calf_feeds()` in `feed.py`
2.  In `formulate_ration()` in `calf_ration.py` (note that at the time of writing this issue the function referenced here is still in a PR under review and not yet in the dev branch)

These methods are currently working because the restricted list of calf feeds established in `feed.py` are overwritten by the methods in `calf_ration.py`. However, sticking with these methods will not allow users to add new calf feeds and we can ideally fix it before it becomes a problem :)

## Possible Solution

I don't have a real solution to suggest but here are a couple of thoughts/intermediate goals:

- The `calf_feeds` list that is set up in `feed.py` should be determined by user inputs
- In addition to the nutrient composition of each calf feed, the calf ration formulation methods needs to group feeds into these three categories: milk, milk replacer, starter
- Perhaps we can add these calf-feed category descriptions to calf feeds in the "feed_type" column of the feed library

## Context (Environment)

Calf diet formulation methods are distinct from the other animal classes because 1) they are baby mammals and you didn't eat the same thing when you were a baby as you do now and 2) the calf's stomach functions closer to how our stomachs function than an adult cow's stomach. These physiological characteristics and some practical farm management factors mean that calf diets have much fewer feeds included in them and that they must have either milk or milk replacer when they are first born and then will gradually reduce the amount of milk they consume as they increase the amount of a grain mixture known as 'starter.'

## Target Deadline

Let's aim for early 2025 but definitely before the end of RuFaS Q2 :)

**Comment by KFosterReed** [comment]
@Isgotkowitz does the Nutrition refresh fix this issue?

**Comment by Isgotkowitz** [comment]
> @Isgotkowitz does the Nutrition refresh fix this issue?

Yes @KFosterReed it does. Calf rations are still extremely limited though. Calf rations:
- must include starter.
- must include either whole milk or milk replacer, or both.

Calves will be fed whatever the user specifies (at the current moment, with user defined rations). But the nutrition requirements of calves are determined by assuming that calves are fed with a combination of whole milk and starter or replacer and starter.

**Comment by KFosterReed** [comment]
This issue was fixed in the Animal Module Refresh
